site/docs/plugins/conversations.md

```diff
@@ -7,1190 +7,1527 @@ next: false
 
 ```ts
-const keyboard = new InlineKeyboard()
-  .text("A", "a").text("B", "b");
-await ctx.reply("A or B?", { reply_markup: keyboard });
-const response = await conversation.waitForCallbackQuery(["a", "b"], {
-  otherwise: (ctx) => ctx.reply("Use the buttons!", { reply_markup: keyboard }),
+await ctx.reply("Send a photo and a caption!");
+const [textContext, photoContext] = await Promise.all([
+  conversation.waitFor(":text"),
+  conversation.waitFor(":photo"),
+]);
+await ctx.replyWithPhoto(photoContext.msg.photo.at(-1).file_id, {
+  caption: textContext.msg.text,
 });
-if (response.match === "a") {
-  // User picked "A".
-} else {
-  // User picked "B".
-}
 ```
 
-:::
-
-Other plugins work fine.
-Some of them just need to be installed differently from how you would usually do it.
-This is relevant for the following plugins:
+In the above example, it does not matter if the user sends a photo or text first.
+Both promises will resolve in the order the user picks to send the two messages the code is waiting for.
+[`Promise.all`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) works normally, it only resolves when all passed promises resolve.
 
-- [hydrate](./hydrate)
-- [i18n](./i18n) and [fluent](./fluent)
-- [emoji](./emoji)
+This can be used to wait for unrelated things, too.
+For example, here is how you install a global exit listener inside the conversation.
 
-They have in common that they all store functions on the context object, which the conversations plugin cannot handle correctly.
-Hence, if you want to combine conversations with one of these grammY plugins, you will have to use special syntax to install the other plugin inside each conversation.
+```ts
+conversation.waitForCommand("exit") // no await!
+  .then(() => conversation.halt());
+```
 
-You can install other plugins inside conversations using `conversation.run`:
+As soon as the conversation [finishes in any way](#exiting-conversations), all pending wait calls will be discarded.
+For example, the following conversation will complete immediately after it was entered, without ever waiting for any updates.
 
 ::: code-group
 
 ```ts [TypeScript]
-async function convo(conversation: MyConversation, ctx: MyContext) {
-  // Install grammY plugins here
-  await conversation.run(plugin());
-  // Continue defining the conversation ...
+async function convo(conversation: Conversation, ctx: Context) {
+  const _promise = conversation.wait() // no await!
+    .then(() => ctx.reply("I will never be sent!"));
+
+  // Conversation is done immediately after being entered.
 }
 ```
 
 ```js [JavaScript]
 async function convo(conversation, ctx) {
-  // Install grammY plugins here
-  await conversation.run(plugin());
-  // Continue defining the conversation ...
+  // Do not await this:
+  const _promise = conversation.wait()
+    .then(() => ctx.reply("I will never be sent!"));
```

Comment:
For consistency with the [previous .ts example](https://github.com/grammyjs/website/pull/1147/files#diff-db0ac3bc98ce61149a6b451516fda7b088009759776851945f38567a4197dc9aR1237):

```suggestion
  const _promise = conversation.wait() // no await!
    .then(() => ctx.reply("I will never be sent!"));
```